internal/api/api.go

```diff
@@ -0,0 +1,180 @@
+package api
+
+import (
+	"context"
+	"errors"
+
+	"github.com/shmel1k/qumomf/internal/storage"
+	"github.com/shmel1k/qumomf/internal/storage/sqlite"
+	"github.com/shmel1k/qumomf/internal/vshard"
+	"github.com/shmel1k/qumomf/internal/vshard/orchestrator"
+)
+
+var (
+	ErrEmptyResult = errors.New("empty result")
+)
+
+type Service interface {
+	ClustersList(context.Context) ([]ClusterInfo, error)
+	ClusterSnapshot(context.Context, string) (vshard.Snapshot, error)
+	ReplicaSet(context.Context, string, vshard.ReplicaSetUUID) (vshard.ReplicaSet, error)
+	Instance(context.Context, string, vshard.ReplicaSetUUID, vshard.InstanceUUID) (vshard.Instance, error)
+	Recoveries(context.Context, string, vshard.ReplicaSetUUID) ([]orchestrator.Recovery, error)
+	Alerts(context.Context) (AlertsResponse, error)
+	ClusterAlerts(context.Context, string) (AlertsResponse, error)
+}
+
+func NewService(db storage.Storage) Service {
+	return &service{
+		db: db,
+	}
+}
+
+type service struct {
+	db storage.Storage
+}
+
+func (s *service) ClustersList(ctx context.Context) ([]ClusterInfo, error) {
+	clustersList, err := s.db.GetClusters(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	resp := make([]ClusterInfo, 0, len(clustersList))
+	for _, cluster := range clustersList {
+		resp = append(resp, ClusterInfo{
```

Comment:
As a user I would like to see the state of the cluster and when it was discovered